### ISSUE TITLE
Impossible to modify or add MQTT v5 properties

### ISSUE THREAD

**Issue by manifest** [vernemq/vernemq#964]
At the moment it is impossible to modify or add MQTT v5 properties from a publish hook.
There is an example:

```erlang
auth_on_publish_m5(
    _Username,  _SubscriberId,
    _QoS, _Topic, Payload, _IsRetain, _Properties) ->

     {ok, #{user_property => [{<<"prop-key">>, <<"prop-val">>}]}}.
```

There is a [PR](https://github.com/vernemq/vernemq/pull/963) with fix.

### Environment

- VerneMQ Version: master (8dfafc0)
- OS: Ubuntu 18.04
- Erlang/OTP version (if building from source): 20.3
- VerneMQ configuration (vernemq.conf) or the changes from the default

### Expected behavior

Modified MQTT v5 properties will be send.

### Actual behaviour

Original MQTT v5 properties are sent.

**Comment by larshesel** [comment]
Thanks for opening the issue and PR - I'll take a look later today

**Comment by larshesel** [comment]
Fixed by #963
